### PR TITLE
Improved weapon dropping

### PR DESCRIPTION
**Pull request by Jagnt**

### CODE REVIEW

File: gamemodes/horde/gamemode/sh_horde.lua [hunk 1]
```diff
@@ -60,6 +60,8 @@ CreateConVar("horde_enable_ammo_gui", 1, FCVAR_ARCHIVE, "Enables ammo UI.")
 CreateConVar("horde_enable_class_models", 1, FCVAR_ARCHIVE, "Enables ammo UI.")
 CreateClientConVar("horde_disable_default_gadget_use_key", 0, FCVAR_ARCHIVE, "Disable default key bind for active gadgets.")
 
+CreateClientConVar("horde_pickup_weapons", "1", true, true, "Allows the pickup of dropped weapons when walking over them.")
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sh_horde.lua:63]: This is running sv & cl, it should only run cl
put this and the convar above it in a if CLIENT block

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 2]
```diff
@@ -237,9 +237,11 @@ hook.Add("PlayerSay", "Horde_Commands", function(ply, input, public)
     elseif text[1] == "!mapconfig" then
         MapConfig(ply)
     elseif text[1] == "!drop" then
-        if ply:GetActiveWeapon() and ply:GetActiveWeapon():IsValid() and ply:GetActiveWeapon().Base == "horde_spell_weapon_base" then
+        local Wep = ply:GetActiveWeapon()
+        if Wep and Wep:IsValid() and Wep:GetActiveWeapon().Base == "horde_spell_weapon_base" then
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_commands.lua:241]: `Wep:GetActiveWeapon().Base` doesn't seem what you want to do

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 3]
```diff
@@ -237,9 +237,11 @@ hook.Add("PlayerSay", "Horde_Commands", function(ply, input, public)
...
             return
         end
+        Wep.firstTaken = false
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_commands.lua:244]: This variable name should be more unique

File: gamemodes/horde/gamemode/sv_economy.lua [hunk 4]
```diff
@@ -417,6 +417,11 @@ end)
 hook.Add("PlayerCanPickupWeapon", "Horde_Economy_Pickup", function (ply, wpn)
     if not ply:IsValid() then return false end
     if ply:IsNPC() then return true end
+    if not wpn.firstTaken then
+        wpn.fistTaken = true  
+        wpn.lastOwner = ply
```
Comment (wrefgtzweve) [inline gamemodes/horde/gamemode/sv_economy.lua:422]: `lastOwner` should be more unique

**Review by wrefgtzweve — APPROVED** [review]

File: gamemodes/horde/gamemode/sv_commands.lua (hunk 2 above)
Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua:241]: this is the default code I just replaced ply:getactiveweapon with a variable

Comment (wrefgtzweve) [reply gamemodes/horde/gamemode/sv_commands.lua:241]: No, you're trying to getactiveweapon a weapon.

Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua:241]: oh I missed that part

File: gamemodes/horde/gamemode/sv_economy.lua (hunk 4 above)
Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_economy.lua]: changed

File: gamemodes/horde/gamemode/sv_commands.lua (hunk 3 above)
Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua]: changed

File: gamemodes/horde/gamemode/sv_commands.lua [hunk 5]
```diff
@@ -237,9 +237,11 @@ hook.Add("PlayerSay", "Horde_Commands", function(ply, input, public)
     elseif text[1] == "!mapconfig" then
         MapConfig(ply)
     elseif text[1] == "!drop" then
-        if ply:GetActiveWeapon() and ply:GetActiveWeapon():IsValid() and ply:GetActiveWeapon().Base == "horde_spell_weapon_base" then
+        local Wep = ply:GetActiveWeapon()
+        if Wep and Wep:IsValid() and Wep.Base == "horde_spell_weapon_base" then
             return
         end
+        Wep.newlyGainedWep = false
```
[the remark this answers is not in the document]
Comment (Jagnt) [reply gamemodes/horde/gamemode/sv_commands.lua]: added

**Pull request merged**